### PR TITLE
feat: Add unit tests for formatDelayText

### PR DESCRIPTION
**Pull request by Elouan1411**

### CODE REVIEW

File: app/src/test/java/com/infomaniak/mail/utils/date/DateFormatUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        
+        every { resources.getQuantityString(R.plurals.daysBeforeSendingReminder, 3, 3) } returns expectedText
+
+        val result = context.formatDelayText(delayMinutes)
+
+        assertEquals(expectedText, result)
+    }
+
+    @Test
+    fun `formatDelayText returns hours when delay is 3 hours`() {
```
Comment (FabianDevel) [inline app/src/test/java/com/infomaniak/mail/utils/date/DateFormatUtilsTest.kt:58]: You must also test limit values, like 31 day, 24 hours, 0, a negative minute delay
All this things will make your tests more reliable, because that's where the bug will appear.

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/test/java/com/infomaniak/mail/utils/date/DateFormatUtilsTest.kt [hunk 2]
```diff
@@ -65,4 +65,52 @@ class DateFormatUtilsTest {
...
+
+        val result = context.formatDelayText(delayMinutes)
+
+        assertEquals(expectedText, result)
+    }
+
+    @Test
+    fun `formatDelayText returns 0 days when delay is 0`() {
+        val delayMinutes = 0
+        val expectedText = "0 days"
```
Comment (FabianDevel) [inline app/src/test/java/com/infomaniak/mail/utils/date/DateFormatUtilsTest.kt]: Shouldn't this be `0 day` ?
Also it doesn't make sens, it should return a fallback or an error

File: app/src/test/java/com/infomaniak/mail/utils/date/DateFormatUtilsTest.kt [hunk 3]
```diff
@@ -65,4 +65,52 @@ class DateFormatUtilsTest {
...
+
+        val result = context.formatDelayText(delayMinutes)
+
+        assertEquals(expectedText, result)
+    }
+
+    @Test
+    fun `formatDelayText returns negative hours when delay is negative minutes`() {
+        val delayMinutes = -60 // -1 hour
+        val expectedText = "-1 hour"
```
Comment (FabianDevel) [inline app/src/test/java/com/infomaniak/mail/utils/date/DateFormatUtilsTest.kt]: This makes no sense, should fallback on a basic value or return an error

Same for day, negative value should not be tolarated

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by FabianDevel — APPROVED** [review]
